### PR TITLE
Test Results Publish - python script updates, making junit optional

### PR DESCRIPTION
**Pull request by KarishmaGhiya**
- For Publishing test results, passing junit xml path as an argument through client.yml
- Updates to client.yml to fix the OS name for publishing test results

@Azure/azure-sdk-eng

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: .azure-pipelines/client.yml [hunk 1]
```diff
@@ -56,17 +56,19 @@ jobs:
     displayName: 'Verify Readmes'
 
   - task: UsePythonVersion@0
+    enabled: false
```
Comment (scbedd) [inline .azure-pipelines/client.yml]: Please remove!

File: .azure-pipelines/client.yml [hunk 2]
```diff
@@ -56,17 +56,19 @@ jobs:
...
     displayName: 'Use Python $(PythonVersion)'
     inputs:
       versionSpec: '$(PythonVersion)'
 
   - script: |
      pip install wheel Jinja2
+    enabled: false
```
Comment (scbedd) [inline .azure-pipelines/client.yml]: Please remove!

File: .azure-pipelines/client.yml [hunk 3]
```diff
@@ -56,17 +56,19 @@ jobs:
...
     displayName: 'Prep Environment'
 
   - task: PythonScript@0
     displayName: 'Analyze dependencies'
-    enabled: true
+    enabled: false
```
Comment (scbedd) [inline .azure-pipelines/client.yml]: Please remove!

File: .azure-pipelines/client.yml [hunk 4]
```diff
@@ -75,12 +77,14 @@ jobs:
     # ComponentGovernance is currently unable to run on pull requests of public projects. Running on non-PR
     # builds should be sufficient.
     condition: and(succeededOrFailed(), ne(variables['Build.Reason'],'PullRequest'))
+    enabled: false
```
Comment (scbedd) [inline .azure-pipelines/client.yml]: REmove.

File: .azure-pipelines/client.yml [hunk 5]
```diff
@@ -75,12 +77,14 @@ jobs:
...
     displayName: 'Component Detection'
 
   - task: PublishBuildArtifacts@1
     # To Publish artifacts generated by the 'Analyze dependencies' task under 'Analyze' job
     displayName: 'Publish Artifacts'
     condition: succeededOrFailed()
+    enabled: false
```
Comment (scbedd) [inline .azure-pipelines/client.yml]: Remove

File: .azure-pipelines/client.yml [hunk 6]
```diff
@@ -128,7 +139,13 @@ jobs:
     displayName: 'Setup and Run Tests'
     inputs:
       scriptPath: 'scripts/devops_tasks/setup_execute_tests.py'
-      arguments: '"$(build_targeting_string)"'
+      arguments: '"$(build_targeting_string)" --junitxml="junit/test-results.xml"'
+    
+  - task: PublishTestResults@2
+    condition: succeededOrFailed()
+    inputs:
+      testResultsFiles: '**/test-*.xml'
```
Comment (scbedd) [inline .azure-pipelines/client.yml]: You set up a specific input above in the python script call, can you reference it specifically as well?

File: scripts/devops_tasks/setup_execute_tests.py [hunk 7]
```diff
@@ -22,12 +22,13 @@
...
 ALLOWED_RETURN_CODES = [] 
 
-def prep_and_run_tests(targeted_packages, python_version):
+def prep_and_run_tests(targeted_packages, python_version, testRes):
     print('running test setup for {}'.format(targeted_packages))
     run_check_call([python_version, dev_setup_script_location, '-p', ','.join([os.path.basename(package_path) for package_path in targeted_packages])], root_dir)
 
     print('Setup complete. Running pytest for {}'.format(targeted_packages))
-    command_array = [python_version, '-m', 'pytest']
+    command_array = [python_version, '-m', 'pytest','--doctest-modules']
```
Comment (scbedd) [inline scripts/devops_tasks/setup_execute_tests.py]: What does the `--doctest-modules` do? Is it required always? Can you add it to to the `testRes` variable by default?

File: scripts/devops_tasks/setup_execute_tests.py [hunk 8]
```diff
@@ -22,12 +22,13 @@
 # evaluating whether we want this or not.
 ALLOWED_RETURN_CODES = [] 
 
-def prep_and_run_tests(targeted_packages, python_version):
+def prep_and_run_tests(targeted_packages, python_version, testRes):
```
Comment (scbedd) [inline scripts/devops_tasks/setup_execute_tests.py]: Python uses `snake_case` for variable names.

File: scripts/devops_tasks/setup_execute_tests.py [hunk 9]
```diff
@@ -43,10 +44,20 @@ def prep_and_run_tests(targeted_packages, python_version):
     parser.add_argument(
         'glob_string',
         nargs='?',
-        help = ('A comma separated list of glob strings that will target the top level directories that contain packages. '
+        help = ('A comma separated list of glob strings that will target the top level directories that contain packages.'
                 'Examples: All = "azure-*", Single = "azure-keyvault", Targeted Multiple = "azure-keyvault,azure-mgmt-resource"'))
 
+     #'--junitxml="junit/test-results.xml"'
```
Comment (scbedd) [inline scripts/devops_tasks/setup_execute_tests.py]: Clean up the default, give a bit more explanation as to the comment above. Like adding an `# example usage: ` on line 50.

File: scripts/devops_tasks/setup_execute_tests.py [hunk 10]
```diff
@@ -43,10 +44,20 @@ def prep_and_run_tests(targeted_packages, python_version):
...
+        #default = 'junit/test-results.xml',
+        help = ('The folder where the test results will be stored in xml format'))
+
     args = parser.parse_args()
     targeted_packages = process_glob_string(args.glob_string, root_dir)
-
-    prep_and_run_tests(targeted_packages, args.python_version)
+    test_results_arg = []
+    if args.test_results:
+        test_results_arg.extend(['--junitxml',args.test_results])
```
Comment (scbedd) [inline scripts/devops_tasks/setup_execute_tests.py]: Nit: Spacing. `, args.test_results`

File: scripts/devops_tasks/setup_execute_tests.py [hunk 11]
```diff
@@ -43,10 +44,20 @@ def prep_and_run_tests(targeted_packages, python_version):
...
+    
+    prep_and_run_tests(targeted_packages, args.python_version,test_results_arg)
```
Comment (scbedd) [inline scripts/devops_tasks/setup_execute_tests.py]: Spacing here too.

File: scripts/devops_tasks/setup_execute_tests.py (hunk 7 above)
Comment (scbedd) [reply scripts/devops_tasks/setup_execute_tests.py]: Spacing too.

**Pull request merged**